grinder/core.py

```diff
@@ -129,6 +131,32 @@ def __init__(
         self.filemanager = GrinderFileManager()
         self.db = GrinderDatabase()
 
+    def onehost(self, host_address: str) -> List[dict]:
+
+        if self.shodan_api_key == "YOUR_DEFAULT_API_KEY":
+            print(f"│ Shodan key is not defined. Skip scan.")
+            print(f"└ ", end="")
+            return []
+
+        dnsResolve = 'https://api.shodan.io/dns/resolve?hostnames=' + host_address + '&key=' + self.shodan_api_key
+        resolved = requests.get(dnsResolve)
+        host_address = resolved.json()[host_address]
+        shodan = ShodanConnector(api_key=self.shodan_api_key)
+        shodan.get_host_info(host_address)
+        shodan_raw_results = shodan.get_onehost_result()
+
+        with open("results_shodan.json", "w") as write_file:
+            json.dump(shodan_raw_results, write_file, indent=2)
```

Comment:
First of all, we have a special module `filemanager.py` for these purposes, so you shouldn't use file logic directly in the core module, only in some special cases (this is not the special case, I suppose).

Reply:
For what reason you use `indent=2`? Why `2`? Did you check other parts of the code? We use the value `4` everywhere.